modmesh/gui/naca.py

```diff
@@ -94,16 +93,49 @@ def calc_naca4_points(number, npoint, open_trailing_edge=False,
     return ret
 
 
-def draw_naca4(world, number, npoint, fac, off_x, off_y, **kw):
+def linear_approximation(world, points):
+    for it in range(points.shape[0] - 1):
+        e = world.add_edge
+        (points[it, 0], points[it, 1], 0, points[it + 1, 0], points[it + 1, 1], 0)
+        print(f"{it}: {e}")
+    return world.nedge
+
+
+def cubic_bezier_approximation(world, points):
+    Vector = core.Vector3dFp64
+
+    for it in range(points.shape[0] - 1):
+        p1 = points[it] + (1 / 3) * (points[it + 1] - points[it])
+        p2 = points[it] + (2 / 3) * (points[it + 1] - points[it])
+
+        b = world.add_bezier(
+            [
+                Vector(points[it, 0], points[it, 1], 0),
+                Vector(p1[0], p1[1], 0),
+                Vector(p2[0], p2[1], 0),
+                Vector(points[it + 1, 0], points[it + 1, 1], 0),
+            ]
+        )
+        b.sample(7)
```

Comment:
Number of sample points should be made as an argument.  The default should be `None`, means automatic determination of the number of linear sample points.  You should come up with a formula to determine a reasonable number of linear sample points.

Reply:
Does this mean that we can call `b.sample()` to sample linear sample points and there is a formula in the [function definition of `sample()`](https://github.com/YenPeiChen07/modmesh/blob/9f18780f1843f1faf2e05922be60674c3c78a353/cpp/modmesh/universe/bezier.hpp#L285) to determine a reasonable number of linear sample points?

Besides, we still can pass an argument to `sample()` function, like we do now with `b.sample(7)`.

Reply:
I mean the number of sample points (passed into `sample()`) should be made as an argument to this function `cubic_bezier_approximate()`.  When the argument is `None`, you use an algorithm to determine what is a proper value to send to `sample()`, instead of hard-coding to 7.

Reply:
Thanks for correcting me. I found I misunderstood what you meant.

It may look like:
```
def cubic_bezier_approximate(world, points, nsample=None):
    Vector = core.Vector3dFp64

    if nsample == None:
        # find a formule to determine a reasonable number of linear sample points.

    for it in range(points.shape[0] - 1):
       ...
       b.sample(nsample)

return world.nbezier
```

Reply:
That is what I meant.